### PR TITLE
Creating cloud function tests

### PR DESCRIPTION
**Pull request by Jeremy-Marchand**

### CODE REVIEW

File: cloud_function/main.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def bearer_oauth(r):
     r.headers["User-Agent"] = "v2RecentSearchPython"
     return r
 
-def connect_to_endpoint(url, params):
+def connect_to_endpoint(url: str, params: str):
     response = requests.get(url, auth=bearer_oauth, params=params)
     print(response.status_code)
```
Comment (antoineeudes) [inline cloud_function/main.py:42]: Mauvaise pratique

Pour un log, utilise une librairie de logging

```
import logging

if __main__=="__name__":
       logging.error("error message")
       logging.info("INFO")
       logging.warning("this is a warning")

```

File: cloud_function/main.py [hunk 2]
```diff
@@ -78,6 +78,7 @@ def main():
     table_id = 'wagon-bootcamp-802.my_dataset.twitter_table'
     data.to_gbq(table_id, if_exists='append')
     print('Tweets successfully merged into the table')
+    return None
```
Comment (antoineeudes) [inline cloud_function/main.py:81]: équivalent à ne rien écrire

File: tests/test_cloud_function.py [hunk 3]
```diff
@@ -20,3 +20,70 @@ def test_last_date_db(mocker):
...
+                'text': '“This weekend won’t be smooth sailing” says Championship Leader Max Verstappen #Formula1 #F1 #F1News #Motorsport https://t.co/ncnzaoqvcd'
+                    }],
+            'meta': {'newest_id': '1544680846883528706',
+                'oldest_id': '1544679599849623552',
+                'result_count': 100,
+                'next_token': 'b26v89c19zqg8o3fpz2m15nlsnmmd4ev3codr28uvwzgd'
+                    }
+            }
+
+json_test_no_next = {'data': [{
```
Comment (antoineeudes) [inline tests/test_cloud_function.py:38]: FAKE_TWITTER_API_RESPONSE_WITH_NO_NEXT

File: cloud_function/main.py [hunk 4]
```diff
@@ -37,7 +37,7 @@ def bearer_oauth(r):
     r.headers["User-Agent"] = "v2RecentSearchPython"
     return r
 
-def connect_to_endpoint(url, params):
+def connect_to_endpoint(url: str, params: str):
```
[the remark this answers is not in the document]
Comment (Jeremy-Marchand) [reply cloud_function/main.py]: `class Tweet(TypedDict):
    lang: str
    created_at: str
    id: str
    text: str

class TweetMeta(TypedDict):
    newest_id: str
    oldest_id: str
    result_count: int
    next_token: NotRequired[str]

class TwitterApiResponse(TypedDict):
    data: list[Tweet]
    meta: TweetMeta

def connect_to_endpoint(url: str, params: dict) -> TwitterApiResponse:`

File: cloud_function/main.py (hunk 1 above)
Comment (Jeremy-Marchand) [reply cloud_function/main.py]:
```
def connect_to_endpoint(url: str, params: dict) -> TwitterApiResponse:
    response = requests.get(url, auth=bearer_oauth, params=params)
    if response.status_code != 200:
        logging.error('Failed to connect, response different from 200')
    else:
        logging.info('Successfully connected to twitter API')
    return response.json()
```

File: cloud_function/main.py (hunk 2 above)
Comment (Jeremy-Marchand) [reply cloud_function/main.py:103]: Done

File: tests/test_cloud_function.py [hunk 5]
```diff
@@ -20,3 +20,70 @@ def test_last_date_db(mocker):
                 return_value = df_test_last_date
                 )
     assert last_date_db() == '2022-06-10T17:57:21Z'
+
+
+json_test_next = {'data': [{
```
[the remark this answers is not in the document]
Comment (Jeremy-Marchand) [reply tests/test_cloud_function.py]: Changed name of variable to FAKE_TWITTER_API_RESPONSE

File: tests/test_cloud_function.py (hunk 3 above)
Comment (Jeremy-Marchand) [reply tests/test_cloud_function.py]: Changed name of variable to FAKE_TWITTER_API_RESPONSE_WITH_NO_NEXT

**Pull request merged**